tests/javascript_tests/test_animate.js

```diff
@@ -99,6 +105,21 @@ $(document).ready(function() {
                   {"x": -0.237661, "y": 0.046053, "z": -0.138136},
                   {"x": -0.276542, "y": -0.144964, "z": 0.066647}],
                   'Fast');
+        equal(director.trajectories.length, 2, 'The number of trajectories is '+
+              'correct');
+    });
+
+    test('Test useless trajectories are removed', function(){
```

Comment:
Could you also check that the number of points in the trajectories is correct? C:2, D:3